src/index/base.h

```diff
@@ -103,6 +103,8 @@ class BaseIndex : public CValidationInterface
     /// Get the name of the index for display in logs.
     virtual const char* GetName() const = 0;
 
+    virtual bool AllowPrune() const = 0;
```

Comment:
a879711 There doesn't seem to be any need for the four declarations of `AllowPrune()` to be protected; they can just be private?

Reply:
Done